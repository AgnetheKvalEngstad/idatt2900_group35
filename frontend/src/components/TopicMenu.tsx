import { Grid2, Button } from "@mui/material";
import MenuBookRoundedIcon from "@mui/icons-material/MenuBookRounded";
import CreateOutlinedIcon from "@mui/icons-material/CreateOutlined";

interface TopicMenuProps {
  difficulty: string;
  index: number;
  onButtonClick: (newIndex: number) => void;
}

export default function TopicMenu({
  difficulty,
  index,
  onButtonClick,
}: TopicMenuProps) {
  const getBackgroundColor = (difficulty: string) => {
    switch (difficulty) {
      case "ingen":
        return "#66DB68";
      case "litt":
        return "#FF9000";
      case "mye":
        return "#E77979";
      default:
        return "#9E9E9E";
    }
  };

  const buttons = [
    { label: "FÃ¸rste oppgave", icon: <MenuBookRoundedIcon /> },
    { label: "Andre oppgave", icon: <MenuBookRoundedIcon /> },
    { label: "Tredje oppgave", icon: <CreateOutlinedIcon /> },
  ];

  return (
    <Grid2
      container={true}
      spacing={3}
      justifyContent="center"
      alignItems="center"
      direction="column"
      className="p-8 pr-10"
    >
      {buttons.map((button, i) => (
        <Button
          key={i}
          startIcon={button.icon}
          variant="contained"
          fullWidth
          size="large"
          style={
            i === index
              ? {
                  backgroundColor: "white",
                  color: "black",
                  border: `2px solid ${getBackgroundColor(difficulty)}`,
                }
              : {
                  backgroundColor: getBackgroundColor(difficulty),
                  color: "black",
                  border: `1px solid black`,
                }
          }
          onClick={() => onButtonClick(i)}
        >
          {button.label}
        </Button>
      ))}
    </Grid2>
  );
}
