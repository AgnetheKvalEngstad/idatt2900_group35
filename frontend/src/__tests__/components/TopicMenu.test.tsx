import { describe, expect, it, beforeEach } from "vitest";
import TopicMenu from "../../components/TopicMenu";
import MenuBookRoundedIcon from "@mui/icons-material/MenuBookRounded";
import CreateOutlinedIcon from "@mui/icons-material/CreateOutlined";
import { JSX } from "react";

describe("Tests for component TopicMenu", () => {
  let difficulty: string;
  let index: number;
  let onButtonClick: (i: number) => number;
  let menuItems: { title: string; icon: JSX.Element }[];
  let topicMenu: JSX.Element;

  beforeEach(() => {
    difficulty = "ingen";
    index = 0;
    onButtonClick = (i: number) => i;
    menuItems = [
      { title: "Subtopic 1", icon: <MenuBookRoundedIcon /> },
      { title: "Subtopic 2", icon: <MenuBookRoundedIcon /> },
      { title: "Subtopic 3", icon: <CreateOutlinedIcon /> },
    ];
    topicMenu = TopicMenu({
      difficulty,
      index,
      onButtonClick,
      menuItems,
    });
  });

  it("should render TopicMenu", () => {
    expect(topicMenu).toMatchSnapshot();
  });

  it("should render three buttons", () => {
    expect(topicMenu.props.children).toHaveLength(3);
  });

  it("should render the first button with the correct icon", () => {
    expect(topicMenu.props.children[0].props.startIcon).toEqual(
      <MenuBookRoundedIcon />
    );
  });

  it("should render the third button with the correct icon", () => {
    expect(topicMenu.props.children[2].props.startIcon).toEqual(
      <CreateOutlinedIcon />
    );
  });
});
