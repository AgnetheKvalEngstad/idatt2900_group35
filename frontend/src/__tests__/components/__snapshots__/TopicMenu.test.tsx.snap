// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for component TopicMenu > should render TopicMenu 1`] = `
<ForwardRef(Grid)
  alignItems="center"
  className="p-8 md:pr-10 w-80"
  container={true}
  direction="column"
  justifyContent="center"
  spacing={3}
>
  <ForwardRef(Button)
    className="hover:shadow-lg hover:scale-105 transition-transform duration-200"
    fullWidth={true}
    onClick={[Function]}
    size="large"
    startIcon={<Memo />}
    style={
      {
        "backgroundColor": "white",
        "border": "2px solid #66DB68",
        "color": "black",
      }
    }
    variant="contained"
  >
    Subtopic 1
  </ForwardRef(Button)>
  <ForwardRef(Button)
    className="hover:shadow-lg hover:scale-105 transition-transform duration-200"
    fullWidth={true}
    onClick={[Function]}
    size="large"
    startIcon={<Memo />}
    style={
      {
        "backgroundColor": "#66DB68",
        "border": "1px solid black",
        "color": "black",
      }
    }
    variant="contained"
  >
    Subtopic 2
  </ForwardRef(Button)>
  <ForwardRef(Button)
    className="hover:shadow-lg hover:scale-105 transition-transform duration-200"
    fullWidth={true}
    onClick={[Function]}
    size="large"
    startIcon={<Memo />}
    style={
      {
        "backgroundColor": "#66DB68",
        "border": "1px solid black",
        "color": "black",
      }
    }
    variant="contained"
  >
    Subtopic 3
  </ForwardRef(Button)>
</ForwardRef(Grid)>
`;
